lib/hsa/mcwamp_hsa.cpp

```diff
@@ -5595,3 +5717,17 @@ std::ostream& operator<<(std::ostream& os, const HSAOp & op)
 //   - store queue, completion signal, other common info.
 
 //   - remove hsaqueeu
+
+////////////////////////////////////////////////////////////////////////////////////////
+// Profiling primitives
+records_dictionary_t* ActivityRecords::_records = NULL;
+activity_callback_t ActivityCallbacks::_fun = NULL;
+void* ActivityCallbacks::_arg = NULL;
+extern "C" __attribute__((visibility("default")))
+size_t HSAOp_get_record_size() { return sizeof(hcc_record_t); }
```

Comment:
it's not a good idea exposing interface in an implementation. I guess the idea for these functions are for upper-layer components to use. In such case a separate header file shall be defined.

Reply:
Yes, I agree, we can talk how to name it and where to put the header.

Reply:
The changes were made against roc-1.8.x branch
To check them against 'master' I will reaply them th enext week I will be back from vacatino